patches/common/chromium/printing.patch

```diff
@@ -0,0 +1,1010 @@
+   PrintPreviewUI* print_preview_ui = GetPrintPreviewUI(ids.ui_id);
+   if (!print_preview_ui)
+     return;
+@@ -318,15 +347,19 @@ void PrintPreviewMessageHandler::OnCompositePdfDocumentDone(
+       base::RefCountedSharedMemoryMapping::CreateFromWholeRegion(region));
+ }
+
++#endif
++
+ bool PrintPreviewMessageHandler::OnMessageReceived(
+     const IPC::Message& message,
+     content::RenderFrameHost* render_frame_host) {
+   bool handled = true;
+   IPC_BEGIN_MESSAGE_MAP_WITH_PARAM(PrintPreviewMessageHandler, message,
+                                    render_frame_host)
++#if 0
+     IPC_MESSAGE_HANDLER(PrintHostMsg_RequestPrintPreview,
+                         OnRequestPrintPreview)
+     IPC_MESSAGE_HANDLER(PrintHostMsg_DidPreviewPage, OnDidPreviewPage)
++#endif
+     IPC_MESSAGE_HANDLER(PrintHostMsg_MetafileReadyForPrinting,
+                         OnMetafileReadyForPrinting)
+     IPC_MESSAGE_UNHANDLED(handled = false)
+@@ -336,10 +369,13 @@ bool PrintPreviewMessageHandler::OnMessageReceived(
+
+   handled = true;
+   IPC_BEGIN_MESSAGE_MAP(PrintPreviewMessageHandler, message)
++#if 0
+     IPC_MESSAGE_HANDLER(PrintHostMsg_DidGetPreviewPageCount,
+                         OnDidGetPreviewPageCount)
++#endif
+     IPC_MESSAGE_HANDLER(PrintHostMsg_PrintPreviewFailed,
+                         OnPrintPreviewFailed)
++#if 0
+     IPC_MESSAGE_HANDLER(PrintHostMsg_DidGetDefaultPageLayout,
+                         OnDidGetDefaultPageLayout)
+     IPC_MESSAGE_HANDLER(PrintHostMsg_PrintPreviewCancelled,
+@@ -348,9 +384,44 @@ bool PrintPreviewMessageHandler::OnMessageReceived(
+                         OnInvalidPrinterSettings)
+     IPC_MESSAGE_HANDLER(PrintHostMsg_SetOptionsFromDocument,
+                         OnSetOptionsFromDocument)
++#endif
+     IPC_MESSAGE_UNHANDLED(handled = false)
+   IPC_END_MESSAGE_MAP()
+   return handled;
+ }
+
++void PrintPreviewMessageHandler::PrintToPDF(
++    const base::DictionaryValue& options,
++    const PrintToPDFCallback& callback) {
++  int request_id;
++  options.GetInteger(printing::kPreviewRequestID, &request_id);
++  print_to_pdf_callback_map_[request_id] = callback;
++
++  content::RenderFrameHost* rfh = web_contents()->GetMainFrame();
++  rfh->Send(new PrintMsg_PrintPreview(rfh->GetRoutingID(), options));
++}
++
++void PrintPreviewMessageHandler::RunPrintToPDFCallback(int request_id,
++                                                       uint32_t data_size,
++                                                       char* data) {
++  DCHECK_CURRENTLY_ON(BrowserThread::UI);
++
++  v8::Isolate* isolate = v8::Isolate::GetCurrent();
++  v8::Locker locker(isolate);
++  v8::HandleScope handle_scope(isolate);
++  if (data) {
++    v8::Local<v8::Value> buffer =
++        node::Buffer::New(isolate, data, static_cast<size_t>(data_size),
++                          &FreeNodeBufferData, nullptr)
++            .ToLocalChecked();
++    print_to_pdf_callback_map_[request_id].Run(v8::Null(isolate), buffer);
```

Comment:
any chance we could make these `OnceCallback`s?

Reply:
Things in the patch in general are things I just pulled incrementally from what I've found in the PRs that are mentioned in the patch description, I guess we should ask the original authors about these, I'm not a 100% confident on answering and fixing these without their approval.